### ISSUE TITLE
Regression in v1.0.7: Fatal Error: Call to a member function mergeData() on array

### ISSUE THREAD

**Issue by ironsmile** [consolidation/Robo#584]
### Steps to reproduce
* Upgraded to version 1.0.7 of robo.
* Run parallel codeception tasks as explained in [this guide](http://codeception.com/docs/12-ParallelExecution#Robo)
* Try to collect the Junit XML results

### Expected behavior
Junit XML files to be happily merged as they have been so far.

### Actual behavior
The robo task for merging Junit XML results fails with

```
PHP Fatal error:  Call to a member function mergeData() on array in /raid1/jenkins/workspace/Inv.bg Selenium Tests/vendor/consolidation/robo/src/Collection/CollectionBuilder.php on line 521
ERROR: Call to a member function mergeData() on array 
in /raid1/jenkins/workspace/Inv.bg Selenium Tests/vendor/consolidation/robo/src/Collection/CollectionBuilder.php:521
```

### System Configuration
OS: Linux
PHP: 5.6

### Comments
This issue is not present in version 1.0.6 of robo. Reverting back to it fixes the problem.

**Comment by Martyrer** [comment]
It is caused by 
```
public function run()
{
        $this->startTimer();
        $result = $this->runTasks();
        $this->stopTimer();
        $result['time'] = $this->getExecutionTime();
        $result->mergeData($this->getState()->getData());
        return $result;
}
```
i suppose it should be changed to

```
public function run()
{
        $this->startTimer();
        $result = $this->runTasks();
        $this->stopTimer();
        $result->accumulateExecutionTime($this->getExecutionTime());
        $result->mergeData($this->getState()->getData());
        return $result;
}
```

**Comment by greg-1-anderson** [comment]
robo-paracept is returning an array instead of a Robo\Result? I don't think that is allowed. All the same, I can check for it and convert.

**Comment by kirilltitov** [comment]
If I recall correctly $this->runTasks returns VOID which is immediately being cast to ARRAY.

**Comment by greg-1-anderson** [comment]
Please test again against #585 and see if it solves your issues.

**Comment by Martyrer** [comment]
the main problem in those 2 lines:
```
$result['time'] = $this->getExecutionTime();
$result->mergeData($this->getState()->getData());
```

$result being casted to Array and that used as an Object these change related to v1.0.7

**Comment by kirilltitov** [comment]
Well, technically $result could implement ArrayAccess interface, so it might be legit :)

**Comment by Martyrer** [comment]
you are right, there is ArrayAccess implementation

**Comment by greg-1-anderson** [comment]
But does it work in your scripts?

**Comment by Martyrer** [comment]
yep, it does work as expected with this fix

**Comment by greg-1-anderson** [comment]
I just released Robo 1.0.8 including the fix that takes care of this regression.
